### DISCUSSION TITLE
SES

### DISCUSSION THREAD

**Discussion by ashconnell** [omigroup/omi-scripting-group#6] [General]
> Reposting here as advised by @cdata

Is there a particular reason that WASM is assumed the solution to UGC?

I’ve been using an implementation of SES (Secure EcmaScript) to compartmentalise UGC with great success. One major benefit of this is that everything runs in the host process, which provides much more capability than a WASM bridge API would ever be able to do, such as using React.

If interested, you can see how I’m using this here: https://twitter.com/ashconnell/status/1470218044681031681?s=21

**Comment by robertlong** [comment]
Hey @ashconnell I started a whole discussion on this with some background and explainer for those who don't already know about SES here: https://github.com/omigroup/omi-wasm-group/discussions/7 Should we continue the discussion over there?

**Comment by KooIaIa** [comment]
Would a engine built to run Secure EcmaScript also probably add a WASM runtime too? Does Secure EcmaScript have a way to connect to WASM like normal EcmaScript?

I feel like something built to support Secure EcmaScript would also probably be able to benefit from supporting WASM too just like browsers benefit from having both. Is there a reason not to support both?

My feeling is that engines which already have a scripting language like C# would be less inclined to support Secure EmcaScript than WASM because their existing engine already covers a lot of the same ground as EmcaScript.

**Reply by ashconnell** [reply]
This is a good point, and probably something I would explore in the future. It’s definitely possible to use both as SES is literally just JavaScript.
